hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestSpark3Catalog.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hudi
+
+import java.sql.{Connection, DriverManager}
+import java.util.Properties
+
+import org.apache.hudi.HoodieSparkUtils
+
+import org.apache.spark.SparkConf
+import org.apache.spark.util.Utils
+
+import java.io.File
+
+class TestSpark3Catalog extends HoodieSparkSqlTestBase {
+
+  val tempDir: File = Utils.createTempDir()
+  val url = s"jdbc:h2:${tempDir.getCanonicalPath};user=testUser;password=testPass"
+
+  override def sparkConf(): SparkConf = {
+    val sparkConf = super.sparkConf()
+    if (HoodieSparkUtils.gteqSpark3_1) {
```

Comment:
Do we really need this conditional here?

Reply:
i think we'd better have this in case it mess up in spark 2.x env.

Reply:
yes for now. But if users run UTs in their local env and some UTs fail, maybe they will be confused.
Like `TestSpark3DDL` and `TestTimeTravelTable`,  these also add this conditional.